tests/components/husqvarna_automower/conftest.py

```diff
@@ -117,33 +117,10 @@ async def listen() -> None:
         await listen_block.wait()
         pytest.fail("Listen was not cancelled!")
 
-    mock = AsyncMock(spec=AutomowerSession)
+    mock = create_autospec(AutomowerSession, instance=True)
```

Comment:
What's the difference?

Reply:
As far as I understand it, it creates a more accurate mock, which can e.g. throw an attribute error, if the arguments for calling the mocked function doesn't match to the original function.

Reply:
I thought that `AsyncMock(spec=..)` already did that trick

Reply:
I tested it and I can't confirm it.

Reply:
Oh right, in my tests i usually have

```py
With patch("....", autospec=True) as mock:
    mock.something
```

Reply:
If mypy wouldn't with https://github.com/home-assistant/core/pull/143113/commits/a6274f1263f06692e7e679eb524e610f8a9a2c9d there would be an AttributeError.
With this https://github.com/home-assistant/core/pull/143113/commits/4a91bb60bdd188896bd49074b34fff9bd18e019e pytest would pass.